core/server/api/v0.1/decorators/urls.js

```diff
@@ -0,0 +1,86 @@
+const urlService = require('../../../services/url');
+const {urlFor, makeAbsoluteUrls} = require('../../../services/url/utils');
+
+const urlsForPost = (post, options) => {
+    post.url = urlService.getUrlByResourceId(post.id);
+
+    if (options.columns && !options.columns.includes('url')) {
+        delete post.url;
+    }
+
+    if (options && options.context && options.context.public && options.absolute_urls) {
+        if (post.feature_image) {
+            post.feature_image = urlFor('image', {image: post.feature_image}, true);
+        }
+
+        if (post.og_image) {
+            post.og_image = urlFor('image', {image: post.og_image}, true);
+        }
+
+        if (post.twitter_image) {
+            post.twitter_image = urlFor('image', {image: post.twitter_image}, true);
+        }
+
+        if (post.html) {
+            post.html = makeAbsoluteUrls(post.html, urlFor('home', true), post.url).html();
+        }
+
+        if (post.url) {
+            post.url = urlFor({relativeUrl: post.url}, true);
+        }
+    }
+
+    if (options && options.withRelated) {
+        options.withRelated.forEach((relation) => {
+            if (relation === 'tags' && post.tags) {
+                post.tags = post.tags.map(tag => urlsForTag(tag, options));
```

Comment:
The computed property primary_tag and primary_author, how is it ensured that they return the correct urls too?
Is it because it's a reference?

Reply:
Good catch with `primary_tag` double checking now. But we could be more explicit here and decorate it as well. What do you think?

Reply:
As for `primary_author`, do you mean ther `author` property?

Reply:
no primary_author

Reply:
So `primary_tag` gets decorated with url as it's passed by reference. My suspicion the same behavior would be with `primari_authors` but can't locate the place where it's assigned to a post :confused:

Reply:
> ut can't locate the place where it's assigned to a post 😕

🤔 in relations/authors

Reply:
Have only found an assignment in `authors.js` serialize method. Investigating the use of it

Reply:
@kirrg001 everything checks out with the decoration of primary_tag/primary_author, added more coverage for that case: https://github.com/TryGhost/Ghost/pull/9917/commits/a0f300e694f86df6c10324b6216fbc07a555ecc1